pallas-miniprotocols/src/blockfetch/codec.rs

```diff
@@ -0,0 +1,72 @@
+use pallas_codec::minicbor::{decode, encode, Decode, Decoder, Encode, Encoder};
+
+use super::Message;
+
+impl Encode<()> for Message {
+    fn encode<W: encode::Write>(
+        &self,
+        e: &mut Encoder<W>,
+        _ctx: &mut (),
+    ) -> Result<(), encode::Error<W::Error>> {
+        match self {
+            Message::RequestRange { range } => {
+                e.array(3)?.u16(0)?;
+                e.encode(&range.0)?;
+                e.encode(&range.1)?;
+                Ok(())
+            }
+            Message::ClientDone => {
+                e.array(1)?.u16(1)?;
+                Ok(())
+            }
+            Message::StartBatch => {
+                e.array(1)?.u16(2)?;
+                Ok(())
+            }
+            Message::NoBlocks => {
+                e.array(1)?.u16(3)?;
+                Ok(())
+            }
+            Message::Block { body } => {
+                e.array(2)?.u16(4)?;
```

Comment:
Missing a CBOR tag here? We decode one in the corresponding decode method

`msgBlock = [4 , #6.24( bytes . cbor block ) ]`

Reply:
This hasn't changed, but maybe we're compensating in other places. Which decode method are you referring to?

Reply:
These are the encode method for the messages, and the corresponding decode method for `Message::Block` decodes a tag. I assume that it hasn't been issue because no one has used outbound server messages